src/basic/terminal-util.c

```diff
@@ -261,8 +263,12 @@ int reset_terminal_fd(int fd, bool switch_to_text) {
         if (switch_to_text)
                 (void) ioctl(fd, KDSETMODE, KD_TEXT);
 
-        /* Enable console unicode mode */
-        (void) ioctl(fd, KDSKBMODE, K_UNICODE);
+        /* Set default keyboard mode */
+        if (read_one_line_file("/sys/module/vt/parameters/default_utf8", &utf8) >= 0 && *utf8 == '1')
```

Comment:
i'd feel more comfortable if you'd use parse_boolean() for parsing what is read here... Yes, it parses a superset of what the file might expose, but it feels more robust and more readable as well